nncf/experimental/onnx/graph/model_transformer.py

```diff
@@ -10,119 +10,121 @@
-                                           value_info=model.graph.value_info)
-            onnx_model = onnx.helper.make_model(graph,
-                                                ir_version=model.ir_version,
-                                                producer_name=model.producer_name,
-                                                producer_version=model.producer_version,
-                                                domain=model.domain,
-                                                model_version=model.model_version,
-                                                doc_string=model.doc_string)
-            if len(model.metadata_props) > 0:
-                values = {p.key: p.value for p in model.metadata_props}
-                onnx.helper.set_model_props(onnx_model, values)
-
-            if len(onnx_model.graph.input) != len(model.graph.input):
-                raise RuntimeError("Input mismatch {} != {}".format(
-                    len(onnx_model.input), len(model.input)))
-            # fix opset import
-            del onnx_model.opset_import[:]
-            for oimp in model.opset_import:
-                op_set = onnx_model.opset_import.add()
-                op_set.domain = oimp.domain
-                op_set.version = oimp.version
-            return onnx_model
-
-        onnx_graph = ONNXGraph(self.transformed_model)
-        nncf_graph = GraphConverter.create_nncf_graph(self.transformed_model)
-        model_outputs = [output.name for output in onnx_graph.get_model_outputs()]
+                output_insert_transformations.append(transformation)
+
+        if quantizer_insert_transformations:
+            self._apply_quantizer_insertion_transformations(quantizer_insert_transformations)
+        if output_insert_transformations:
+            self._apply_output_insertion_transformations(output_insert_transformations)
+
+    def _apply_output_insertion_transformations(self, transformations: List[ONNXOutputInsertionCommand]) -> None:
+        """
+        Applies incoming transformations to the model
+
+        :param transformations: list of the ONNXOutputInsertionCommand transformations
+        """
+        backend_graph = BackendGraphFactory.create(self._model)
+        nncf_graph = NNCFGraphFactory.create(self._model)
+        model_outputs = self._get_model_outputs(backend_graph)
+        extra_model_outputs = self._get_extra_model_outputs(nncf_graph,
+                                                            backend_graph,
+                                                            transformations)
+
+        model_with_intermediate_outputs = self._insert_outputs(self._model,
+                                                               outputs=[*extra_model_outputs,
+                                                                        *model_outputs])
+        self._model = model_with_intermediate_outputs
+
+    def _get_model_outputs(self, onnx_graph: ONNXGraph) -> List:
```

Comment:
Please add such function to the `ONNXGraph` or use directly `[output.name for output in onnx_graph.get_model_outputs()]`

Reply:
Done